[Report] Hardware Tessellation test on android failed.

Just simple report.
I tried auto-built v1.3-321-gce2feb2 but it's failed with Hardware Tessellation.
I hope to fix it but I don't know how to fix and debug with android.

Tested bezier in pspgl on ZenFone 5(Android 4.4.2)

![screenshot_2017-01-25-05-13-38](https://cloud.githubusercontent.com/assets/1607422/22273572/dbcb90ae-e2e4-11e6-88c6-79be8bacaa97.jpg)

So you actually checked game which uses this code or the test homebrew?

Meanwhile I found a way to reproduce the problem on pc;p linux under virtualbox - yay for poor gpu/drivers.

Log from test homebrew:
```
51:02:871 user_main    E[G3D]: GLES/ShaderManagerGLES.cpp:73 Error in shader compilation!

51:02:875 user_main    E[G3D]: GLES/ShaderManagerGLES.cpp:74 Info log: 0:92(18): error: `gl_InstanceID' undeclared
0:92(14): error: no matching function for call to `mod(error, int)'; candidates are:
0:92(14): error:    float mod(float, float)
0:92(14): error:    vec2 mod(vec2, float)
0:92(14): error:    vec3 mod(vec3, float)
0:92(14): error:    vec4 mod(vec4, float)
0:92(14): error:    vec2 mod(vec2, vec2)
0:92(14): error:    vec3 mod(vec3, vec3)
0:92(14): error:    vec4 mod(vec4, vec4)
0:92(10): error: cannot construct `int' from a non-numeric data type
0:93(11): error: `gl_InstanceID' undeclared
0:93(11): error: operands to arithmetic operators must be numeric


51:02:876 user_main    E[G3D]: GLES/ShaderManagerGLES.cpp:75 Shader source:
#version 130
#define lowp
#define mediump
#define highp
attribute vec3 position;
attribute mediump vec3 normal;
attribute lowp vec4 color0;
uniform mat4 u_proj;
uniform mat4 u_world;
uniform mat4 u_view;
uniform vec3 u_lightpos0;
uniform mediump vec3 u_lightatt0;
uniform mediump vec3 u_lightdir0;
uniform mediump float u_lightangle0;
uniform mediump float u_lightspotCoef0;
uniform lowp vec3 u_lightambient0;
uniform lowp vec3 u_lightdiffuse0;
uniform lowp vec3 u_lightspecular0;
uniform vec3 u_lightpos1;
uniform mediump vec3 u_lightatt1;
uniform mediump vec3 u_lightdir1;
uniform mediump float u_lightangle1;
uniform mediump float u_lightspotCoef1;
uniform lowp vec3 u_lightambient1;
uniform lowp vec3 u_lightdiffuse1;
uniform lowp vec3 u_lightspecular1;
uniform vec3 u_lightpos2;
uniform mediump vec3 u_lightatt2;
uniform mediump vec3 u_lightdir2;
uniform mediump float u_lightangle2;
uniform mediump float u_lightspotCoef2;
uniform lowp vec51:02:878 user_main    E[G3D]: GLES/ShaderManagerGLES.cpp:845 Shader compilation failed, falling back to software transform
```

So mod() fails and expects floats since as with my edit above integer math requires GL_EXT_gpu_shader4, so that's one thing we need to check and enable. Not doing that could probably be the reason why it failed for the person with nvidia gpu on win 10 reporting in merge thread. Another problem is gl_InstanceID which requires gles 3.00 or if someone really cares, it could also work on older iOS with some [apple stuff](http://stackoverflow.com/questions/25387959/instanced-drawing-with-opengl-es-2-0-on-ios), probably not worth.

 Guess both of those could be supported on new higher end smartphones like mentioned GS6.

--- a/GPU/GLES/GPU_GLES.cpp
+++ b/GPU/GLES/GPU_GLES.cpp
@@ -600,7 +600,7 @@ void GPU_GLES::CheckGPUFeatures() {
 	if (maxVertexTextureImageUnits >= 3) // At least 3 for hardware tessellation
 		features |= GPU_SUPPORTS_VERTEX_TEXTURE_FETCH;
 
-	if (gl_extensions.ARB_texture_float || gl_extensions.OES_texture_float)
+	if (gl_extensions.ARB_texture_float || gl_extensions.OES_texture_float || gl_extensions.OES_texture_half_float)
 		features |= GPU_SUPPORTS_TEXTURE_FLOAT;
 
 	// If we already have a 16-bit depth buffer, we don't need to round.

--- a/UI/GameSettingsScreen.cpp
+++ b/UI/GameSettingsScreen.cpp
@@ -83,7 +83,7 @@ bool CheckSupportInstancedTessellation() {
 	glGetIntegerv(GL_MAX_VERTEX_TEXTURE_IMAGE_UNITS, &maxVertexTextureImageUnits);
 	bool vertexTexture = maxVertexTextureImageUnits >= 3; // At least 3 for hardware tessellation
 	bool instanceRendering = gl_extensions.EXT_gpu_shader4;
-	bool textureFloat = gl_extensions.ARB_texture_float || gl_extensions.OES_texture_float;
+	bool textureFloat = gl_extensions.ARB_texture_float || gl_extensions.OES_texture_float || gl_extensions.OES_texture_half_float;
 
 	return instanceRendering && vertexTexture && textureFloat;
 }

--- a/ext/native/gfx_es2/gpu_features.cpp
+++ b/ext/native/gfx_es2/gpu_features.cpp
@@ -280,6 +280,7 @@ void CheckGLExtensions() {
 		gl_extensions.NV_shader_framebuffer_fetch = strstr(extString, "GL_NV_shader_framebuffer_fetch") != 0;
 		gl_extensions.ARM_shader_framebuffer_fetch = strstr(extString, "GL_ARM_shader_framebuffer_fetch") != 0;
 		gl_extensions.OES_texture_float = strstr(extString, "GL_OES_texture_float") != 0;
+		gl_extensions.OES_texture_half_float = strstr(extString, "GL_OES_texture_half_float") != 0;
 
 #if defined(__ANDROID__)
 		// On Android, incredibly, this is not consistently non-zero! It does seem to have the same value though.

--- a/ext/native/gfx_es2/gpu_features.cpp
+++ b/ext/native/gfx_es2/gpu_features.cpp
@@ -279,7 +279,7 @@ void CheckGLExtensions() {
 		gl_extensions.EXT_shader_framebuffer_fetch = strstr(extString, "GL_EXT_shader_framebuffer_fetch") != 0;
 		gl_extensions.NV_shader_framebuffer_fetch = strstr(extString, "GL_NV_shader_framebuffer_fetch") != 0;
 		gl_extensions.ARM_shader_framebuffer_fetch = strstr(extString, "GL_ARM_shader_framebuffer_fetch") != 0;
-		gl_extensions.OES_texture_float = strstr(extString, "OES_texture_float") != 0;
+		gl_extensions.OES_texture_float = strstr(extString, "GL_OES_texture_float") != 0;
 
 #if defined(__ANDROID__)
 		// On Android, incredibly, this is not consistently non-zero! It does seem to have the same value though.

--- a/ext/native/gfx_es2/gpu_features.h
+++ b/ext/native/gfx_es2/gpu_features.h
@@ -48,6 +48,7 @@ struct GLExtensions {
 	bool OES_vertex_array_object;
 	bool OES_copy_image;
 	bool OES_texture_float;
+	bool OES_texture_half_float;
 
 	// ARB
 	bool ARB_framebuffer_object;

